RIOT-OS/RIOT
# drivers/ds18: Add Maxim Integrated 1-Wire temperature sensor driver

### Contribution description
This PR is based on #6512.

What this PR adds:
- Fixes the temperature reading problem reported in #6512.
- Fixes some type errors on the API.
- Adds a timeout to the ```ds18_read_bit``` function.
- Test application for the driver.

### Testing procedure
1. Connect the sensor to a free GPIO on the board.
2. Define  ```DS18_PARAM_PIN``` and ```DS18_PARAM_PULL``` accordingly to your board.
3. Run the test application with something like: ```BOARD=lobaro-lorabox make flash term -C tests/driver_ds18```.
4. You should see an output similar to:

```
# DS18B20 test application
# 
# +------------Initializing------------+
# 
# +--------Starting Measurements--------+
# Temperature [ºC]: 25.87
# +-------------------------------------+
```

### Issues/PRs references
Based on #6512.
